CI regression: Notepad UIA detection returns only 'vision' method (re-open of #570)

## Problem
CI desktop tests `test_detect_notepad_has_uia` and `test_detect_notepad_best_method` are failing again. The detection chain for Notepad returns only `vision` method instead of including `uia`.

This was previously fixed in #570 but has regressed.

## Test Output
```
AssertionError: Notepad should support UIA, got methods: ['vision']
assert 'uia' in ['vision']

AssertionError: Expected UIA or MSAA for Win32 app, got vision
assert 'vision' in ('uia', 'msaa')
```

## Note
Interestingly, `naturo see --app Notepad` works fine and returns UIA elements in manual testing. This suggests the detection chain probing in the test context may differ from the CLI context.

## CI Results (commit 7a52fc1)
- 2 FAILED: test_detect_notepad_has_uia, test_detect_notepad_best_method
- 3 FAILED (e2e): test_window_minimize_restore_cycle (timeout), test_notepad_open_type_close, test_multi_window_list_all
- 17 PASSED (desktop/integration)
- 7 PASSED (e2e)

## Environment
- Windows 11 Pro 10.0.26200
- naturo v0.3.0
- Runner: NATUROBOT

**[QA-Mariana]**

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -36,6 +36,16 @@ if _CI_WINDOWS:
 
     class _SkippingNaturoCore:
         def __init__(self, *args, **kwargs):
+            import threading
+            if threading.current_thread() is not threading.main_thread():
+                # (#683) pytest.skip() fails in daemon threads (no pytest
+                # request context).  The detection chain runs probes in
+                # daemon threads, so raise a regular exception instead —
+                # the probe's exception handler logs and returns None.
+                raise RuntimeError(
+                    "NaturoCore() blocked on CI Windows (no desktop session, "
+                    "daemon thread context)"
+                )
             pytest.skip(
                 "NaturoCore() skipped on CI Windows (no desktop session). "
                 "Add @pytest.mark.desktop if this test requires a real desktop.",

